app/src/main/java/com/android/solvit/provider/ui/profile/ModifyProviderInformation.kt

```diff
@@ -0,0 +1,350 @@
+  val okNewCompanyName = newCompanyName.length >= 2 && newCompanyName.isNotBlank()
+  var newProfession by remember { mutableStateOf(provider.service) }
+
+  var newPhoneNumber by remember { mutableStateOf(provider.phone) }
+  val okNewPhoneNumber =
+      newPhoneNumber.isNotBlank() &&
+          newPhoneNumber.all { it.isDigit() || it == '+' } &&
+          newPhoneNumber.length > 6
+
+  val newLocation by remember { mutableStateOf(provider.location.name) }
+
+  var showDropdown by remember { mutableStateOf(false) }
+
+  val locationSuggestions by
+      locationViewModel.locationSuggestions.collectAsState(initial = emptyList<Location?>())
+  var selectedLocation by remember { mutableStateOf<Location?>(provider.location) }
+  val okNewLocation = selectedLocation != null
+
+  var newLanguage by remember { mutableStateOf(provider.languages) }
+
+  val allIsGood = okNewCompanyName && okNewPhoneNumber && okNewLocation
+
+  CustomOutlinedTextField(
+      value = newCompanyName,
+      onValueChange = { newCompanyName = it },
+      label = "Provider Name",
+      placeholder = "Enter your new company name",
+      isValueOk = okNewCompanyName,
+      leadingIcon = Icons.Default.AccountCircle,
+      leadingIconDescription = "Company Name Icon",
+      testTag = "newProviderCompanyNameInputField",
+      errorMessage = "Your company name must have at least 2 characters",
+      errorTestTag = "providerNameErrorMessage")
+
+  Spacer(modifier = Modifier.height(10.dp))
+
+  ServiceDropdownMenu(selectedService = newProfession, onServiceSelected = { newProfession = it })
+
+  Spacer(modifier = Modifier.height(10.dp))
+
+  CustomOutlinedTextField(
+      value = newPhoneNumber,
+      onValueChange = { newPhoneNumber = it },
+      label = "Phone number",
+      placeholder = "Enter your new phone number",
+      isValueOk = okNewPhoneNumber,
+      leadingIcon = Icons.Default.Phone,
+      leadingIconDescription = "Phone Number Icon",
+      testTag = "newPhoneNumberInputField",
+      errorMessage = "Your phone number name must have at least 7 characters",
+      errorTestTag = "newPhoneNumberErrorMessage")
+
+  Spacer(modifier = Modifier.height(10.dp))
+
+  LocationDropdown(
+      locationQuery = newLocation,
+      onLocationQueryChange = { locationViewModel.setQuery(it) },
+      showDropdownLocation = showDropdown,
+      onShowDropdownLocationChange = { showDropdown = it },
+      locationSuggestions = locationSuggestions.filterNotNull(),
+      onLocationSelected = { selectedLocation = it },
+      requestLocation = null,
+      backgroundColor = colorScheme.background,
+      isValueOk = okNewLocation,
+      testTag = "newLocationInputField")
+
+  LanguageDropdownMenu(
+      selectedLanguages = newLanguage,
+      onLanguageSelected = { language, isChecked ->
+        newLanguage =
+            if (isChecked) {
+              newLanguage + language
+            } else {
+              newLanguage - language
+            }
+      })
+
```

Comment:
Same comment concerning the background color. If I can add something, I do not like the appearance of the different looking fields intertwined. Or all the fields should look the same, or they should be grouped by appearance.